pkg/ebpf/bytecode/manager.go

```diff
@@ -56,29 +56,29 @@ func NewManager(closedHandler *PerfHandler) *manager.Manager {
 			},
 		},
 		Probes: []*manager.Probe{
-			{Section: string(TCPSendMsg)},
-			{Section: string(TCPSendMsgPre410)},
+			{Section: string(TCPSendMsg), Optional: true},
```

Comment:
I don't believe any of our probes are optional. If we put it into the activated list, we need it to start.

Reply:
The `Optional` flag feels like a mixing of concerns too. My understanding was that this list was a declarative approach to what was contained within an ELF file. Any of the runtime concerns about what probes to load/enable was delegated to the `Manager.Options` in `Manager.InitWithOptions`.

Reply:
You're right. Let me try to come up with something better on the eBPF lib side and I'll let you know when this PR is ready for a review again 👍

Reply:
As for `TCPSendMsg` and `TCPSendMsgPre410` I do believe that you always need at least one, but never the 2 at the same time (since their activation is based on the kernel version). This is a perfect example of a `OneOf` probe selector.